tpgtools/main.go

```diff
@@ -183,6 +167,11 @@ func loadAndModelResources() map[Version][]*Resource {
 					}
 				}
 
+				overrides := loadOverrides(packagePath, f.Name())
+				if _, ok := productOverrides[productMetadata.PackageName]; !ok {
```

Comment:
This may not work correctly with per-service filtering

So, the tricky part about per-product information is that it is typically used outside of the scope of the resource being generated, and must be present for the provider as a whole to compile. Check out the `sFilter` field in this class, as it prevents this code from being hit for products that are not included in the product filter command line argument. If this code isn't hit we won't define required fields that must always be present in the provider, causing errors

We will also run into issues if we try to move over `provider.go.erb` at some point, which must have the full list of generated resources regardless of which product/resource is being generated in a given run.

See https://github.com/GoogleCloudPlatform/magic-modules/blob/master/mmv1/compiler.rb for how this works in MMv1.

Reply:
I see what you mean, so we will have to separate out products into their own loop